spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/GqlDatastoreQuery.java

```diff
@@ -118,7 +117,28 @@ public Object execute(Object[] parameters) {
 		List<T> rawResult = found == null ? Collections.emptyList()
 				: StreamSupport.stream(found.spliterator(), false)
 						.collect(Collectors.toList());
-		return applyProjection(rawResult);
+		Object result;
+		if (isCountQuery()) {
+			result = rawResult.size();
```

Comment:
Cloud Spanner has the `COUNT` aggregate function. So, I don't think we need to do anything special here.
If someone provides `SELECT count(*) from users`, this code will always return 1. That's not right. We should just allow mapping to a singe POJO and error out if the result set has more than 1 row.

See: https://cloud.google.com/spanner/docs/functions-and-operators#aggregate-functions

Reply:
This one is datastore though !

Reply:
Oops :)